ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -165,6 +169,24 @@ class ( -- Requirements on the ledger state itself
 applyChainTick :: IsLedger l => LedgerCfg l -> SlotNo -> l -> Ticked l
 applyChainTick = lrResult ..: applyChainTickLedgerResult
 
+-- | Handler for ledger events
+newtype LedgerEventHandler m l blk =
+  LedgerEventHandler
+    { handleLedgerEvent
+        :: ChainHash blk -- Previous block header hash
+        -> HeaderHash l -- Block header hash of the applied block
+        -> SlotNo -- Slot number of the applied block
+        -> BlockNo -- Applied block number
+        -> [AuxLedgerEvent l] -- Resulting 'AuxLedgerEvent's after applying `applyBlock`.
```

Comment:
@KtorZ Changed it to a list so that clients and know if they missed an event or if a rollback occured.